cmd/argo/commands/client/conn.go

```diff
@@ -16,13 +16,16 @@ var argoServer string
 
 var Config clientcmd.ClientConfig
 
+var ExplicitPath string
+
 func AddKubectlFlagsToCmd(cmd *cobra.Command) {
 	// The "usual" clientcmd/kubectl flags
 	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
 	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
 	overrides := clientcmd.ConfigOverrides{}
 	kflags := clientcmd.RecommendedConfigOverrideFlags("")
-	cmd.PersistentFlags().StringVar(&loadingRules.ExplicitPath, "kubeconfig", "", "Path to a kube config. Only required if out-of-cluster")
+	cmd.PersistentFlags().StringVar(&ExplicitPath, "kubeconfig", "", "Path to a kube config. Only required if out-of-cluster")
```

Comment:
This help string is not very clear, "Only required if out-of-cluster" is ambiguous. To someone who is reading the command help with no context this won't be helpful.

Reply:
I think this message took from kube-client. Argo CLI is inheriting most of `kube client` functionality and code. 
https://github.com/kubernetes/client-go/pull/29/files/cf262eae7dee39cab1d8c49fba760d74678f9e97